AppInsights Outputs to stderr for react-native commands When Network Flaky or Unavailable

Showed up during CI when running `react-native bundle` https://dev.azure.com/ms/react-native-windows/_build/results?buildId=121398&view=logs&j=2aef21c9-0610-5c30-26c1-4f046de16038&s=d654deb9-056d-50a2-1717-90c08683d50a&t=93f898e5-507a-5eb1-f73b-c0c953f43c13&l=11471

It looks like telemetry client init does network operations and will take things down and error out if it fails. Since this is done in telemetry setup, it's likely this also happens with the no-telemetry option.

```
MakeTestBundles:
  Creating directory "D:\a\1\s\\vnext\\target\x86\Debug\React.Windows.Desktop.ABITests\".
  npx --no-install react-native bundle --platform windows --entry-file D:\a\1\s\vnext\Microsoft.ReactNative.IntegrationTests\AddValues.js --bundle-output D:\a\1\s\\vnext\\target\x86\Debug\React.Windows.Desktop.ABITests\AddValues.bundle
                   Welcome to React Native!
                  Learn once, write anywhere
  
  
  ApplicationInsights:CorrelationIdManager [
##[error]EXEC(0,0): Error : getaddrinfo ENOTFOUND dc.services.visualstudio.com
EXEC : error : getaddrinfo ENOTFOUND dc.services.visualstudio.com [D:\a\1\s\vnext\Desktop.ABITests\React.Windows.Desktop.ABITests.vcxproj]
        at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26) {
      errno: 'ENOTFOUND',
      code: 'ENOTFOUND',
      syscall: 'getaddrinfo',
      hostname: 'dc.services.visualstudio.com'
    }
  ]
  info Writing bundle output to:, D:\a\1\s\\vnext\\target\x86\Debug\React.Windows.Desktop.ABITests\AddValues.bundle
  info Done writing bundle output
  warn Assets destination folder is not set, skipping...
##[error]vnext\Microsoft.ReactNative.IntegrationTests\TestBundle.targets(9,3): Error MSB3073: The command "npx --no-install react-native bundle --platform windows --entry-file D:\a\1\s\vnext\Microsoft.ReactNative.IntegrationTests\AddValues.js --bundle-output D:\a\1\s\\vnext\\target\x86\Debug\React.Windows.Desktop.ABITests\AddValues.bundle" exited with code -1.


```

--- a/change/@react-native-windows-cli-1f2b01ad-b55e-4589-b852-101ea9e81fe5.json
+++ b/change/@react-native-windows-cli-1f2b01ad-b55e-4589-b852-101ea9e81fe5.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Don't initialize AppInsights when passing no-telemetry",
+  "packageName": "@react-native-windows/cli",
+  "email": "asklar@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/change/@react-native-windows-telemetry-2020-11-09-01-02-17-silenceWarnAI.json
+++ b/change/@react-native-windows-telemetry-2020-11-09-01-02-17-silenceWarnAI.json
@@ -1,0 +1,8 @@
+{
+  "type": "prerelease",
+  "comment": "Silence warnings from appinsights in production",
+  "packageName": "@react-native-windows/telemetry",
+  "email": "asklar@microsoft.com",
+  "dependentChangeType": "patch",
+  "date": "2020-11-09T09:02:16.892Z"
+}

--- a/change/react-native-windows-init-c8db854d-d6e4-49a8-a3bd-c5fe0ed5427b.json
+++ b/change/react-native-windows-init-c8db854d-d6e4-49a8-a3bd-c5fe0ed5427b.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Don't initialize AppInsights when passing no-telemetry",
+  "packageName": "react-native-windows-init",
+  "email": "asklar@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/@react-native-windows/cli/src/index.ts
+++ b/packages/@react-native-windows/cli/src/index.ts
@@ -6,7 +6,7 @@
 
 import * as fs from 'fs';
 import * as path from 'path';
-import {telemetryClient} from '@react-native-windows/telemetry';
+import {Telemetry} from '@react-native-windows/telemetry';
 
 import {
   copyProjectTemplateAndReplace,
@@ -100,7 +100,7 @@ export async function generateWindows(
     );
   } catch (e) {
     error = e;
-    telemetryClient.trackException({exception: error});
+    Telemetry.client?.trackException({exception: error});
     throw e;
   } finally {
     if (options.telemetry && !process.env.AGENT_NAME) {
@@ -122,7 +122,7 @@ export async function generateWindows(
         cliVersion = rnwCliPkgJson.version;
       } catch {}
       const optScrubbed = scrubOptions(options);
-      telemetryClient.trackEvent({
+      Telemetry.client?.trackEvent({
         name: 'generate-windows',
         properties: {
           error: error,
@@ -132,7 +132,7 @@ export async function generateWindows(
         },
       });
 
-      telemetryClient.flush();
+      Telemetry.client?.flush();
     }
   }
 }

--- a/packages/@react-native-windows/cli/src/runWindows/runWindows.ts
+++ b/packages/@react-native-windows/cli/src/runWindows/runWindows.ts
@@ -6,7 +6,7 @@
 
 import * as fs from 'fs';
 import {
-  telemetryClient,
+  Telemetry,
   isMSFTInternal,
   getDiskFreeSpace,
 } from '@react-native-windows/telemetry';
@@ -72,7 +72,9 @@ async function runWindows(
     if (options.logging) {
       console.log('Disabling telemetry');
     }
-    telemetryClient.config.disableAppInsights = true;
+    Telemetry.disable();
+  } else {
+    Telemetry.setup();
   }
 
   // https://github.com/yarnpkg/yarn/issues/8334 - Yarn on Windows breaks apps that read from the environment variables
@@ -93,7 +95,7 @@ async function runWindows(
       sdks.forEach(version => console.log('    ' + version));
       return;
     } catch (e) {
-      telemetryClient.trackException({exception: e});
+      Telemetry.client?.trackException({exception: e});
       newError('Unable to print environment info.\n' + e.toString());
       return setExitProcessWithError(options.logging);
     }
@@ -103,11 +105,11 @@ async function runWindows(
   try {
     await runWindowsInternal(args, config, options);
   } catch (e) {
-    telemetryClient.trackException({exception: e});
+    Telemetry.client?.trackException({exception: e});
     runWindowsError = e;
     return setExitProcessWithError(options.logging);
   } finally {
-    telemetryClient.trackEvent({
+    Telemetry.client?.trackEvent({
       name: 'run-windows',
       properties: {
         release: options.release,
@@ -145,7 +147,7 @@ async function runWindows(
         cpus: cpus().length,
       },
     });
-    telemetryClient.flush();
+    Telemetry.client?.flush();
   }
 }
 

--- a/packages/@react-native-windows/telemetry/src/index.ts
+++ b/packages/@react-native-windows/telemetry/src/index.ts
@@ -4,8 +4,4 @@
  * @format
  */
 
-export {
-  Telemetry,
-  isMSFTInternal,
-  getDiskFreeSpace,
-} from './telemetry';
+export {Telemetry, isMSFTInternal, getDiskFreeSpace} from './telemetry';

--- a/packages/@react-native-windows/telemetry/src/index.ts
+++ b/packages/@react-native-windows/telemetry/src/index.ts
@@ -5,7 +5,7 @@
  */
 
 export {
-  telClient as telemetryClient,
+  Telemetry,
   isMSFTInternal,
   getDiskFreeSpace,
 } from './telemetry';

--- a/packages/@react-native-windows/telemetry/src/telemetry.ts
+++ b/packages/@react-native-windows/telemetry/src/telemetry.ts
@@ -31,7 +31,9 @@ export class Telemetry {
       Telemetry.client.commonProperties.isTest = process.env.RNW_CLI_TEST;
     }
     if (!Telemetry.client.commonProperties.sessionId) {
-      Telemetry.client.commonProperties.sessionId = randomBytes(16).toString('hex');
+      Telemetry.client.commonProperties.sessionId = randomBytes(16).toString(
+        'hex',
+      );
       Telemetry.client.addTelemetryProcessor(sanitizeEnvelope);
     }
   }

--- a/packages/@react-native-windows/telemetry/src/telemetry.ts
+++ b/packages/@react-native-windows/telemetry/src/telemetry.ts
@@ -9,12 +9,34 @@ import {randomBytes} from 'crypto';
 import * as appInsights from 'applicationinsights';
 import {execSync} from 'child_process';
 
-if (!process.env.RNW_CLI_TEST) {
-  appInsights.Configuration.setInternalLogging(false, false);
-}
+export class Telemetry {
+  static client?: appInsights.TelemetryClient | undefined = undefined;
+  static disable() {
+    if (Telemetry.client) {
+      Telemetry.client.config.disableAppInsights = true;
+    }
+    Telemetry.shouldDisable = true;
+  }
+  static setup() {
+    if (!process.env.RNW_CLI_TEST) {
+      appInsights.Configuration.setInternalLogging(false, false);
+    }
+    appInsights.setup('795006ca-cf54-40ee-8bc6-03deb91401c3');
+    Telemetry.client = appInsights.defaultClient;
 
-appInsights.setup('795006ca-cf54-40ee-8bc6-03deb91401c3');
-export const telClient = appInsights.defaultClient;
+    if (Telemetry.shouldDisable) {
+      Telemetry.disable();
+    }
+    if (process.env.RNW_CLI_TEST) {
+      Telemetry.client.commonProperties.isTest = process.env.RNW_CLI_TEST;
+    }
+    if (!Telemetry.client.commonProperties.sessionId) {
+      Telemetry.client.commonProperties.sessionId = randomBytes(16).toString('hex');
+      Telemetry.client.addTelemetryProcessor(sanitizeEnvelope);
+    }
+  }
+  static shouldDisable: boolean = false;
+}
 
 function getAnonymizedPath(filepath: string): string {
   const projectRoot = process.cwd().toLowerCase();
@@ -104,15 +126,6 @@ export function sanitizeEnvelope(envelope: any /*context: any*/): boolean {
   }
   delete envelope.tags['ai.cloud.roleInstance'];
   return true;
-}
-
-if (process.env.RNW_CLI_TEST) {
-  telClient.commonProperties.isTest = process.env.RNW_CLI_TEST;
-}
-
-if (!telClient.commonProperties.sessionId) {
-  telClient.commonProperties.sessionId = randomBytes(16).toString('hex');
-  telClient.addTelemetryProcessor(sanitizeEnvelope);
 }
 
 export function isMSFTInternal(): boolean {

--- a/packages/@react-native-windows/telemetry/src/telemetry.ts
+++ b/packages/@react-native-windows/telemetry/src/telemetry.ts
@@ -9,6 +9,10 @@ import {randomBytes} from 'crypto';
 import * as appInsights from 'applicationinsights';
 import {execSync} from 'child_process';
 
+if (!process.env.RNW_CLI_TEST) {
+  appInsights.Configuration.setInternalLogging(false, false);
+}
+
 appInsights.setup('795006ca-cf54-40ee-8bc6-03deb91401c3');
 export const telClient = appInsights.defaultClient;
 

--- a/packages/@react-native-windows/telemetry/src/test/sanitize.test.ts
+++ b/packages/@react-native-windows/telemetry/src/test/sanitize.test.ts
@@ -5,30 +5,31 @@
  * @format
  */
 
-import * as telemetry from '../telemetry';
+import {Telemetry, sanitizeMessage, sanitizeEnvelope, sanitizeFrame} from '../telemetry';
 import * as appInsights from 'applicationinsights';
 import {basename} from 'path';
 
-telemetry.telClient.config.disableAppInsights = true;
+Telemetry.setup();
+Telemetry.client!.config.disableAppInsights = true;
 
 test('Sanitize message, no-op', () => {
   // do stuff
-  expect(telemetry.sanitizeMessage('')).toEqual('');
-  expect(telemetry.sanitizeMessage('some text')).toEqual('some text');
+  expect(sanitizeMessage('')).toEqual('');
+  expect(sanitizeMessage('some text')).toEqual('some text');
 });
 
 test('Sanitize message, project_dir', () => {
   expect(
-    telemetry.sanitizeMessage(`this is the cwd: '${process.cwd()}'`),
+    sanitizeMessage(`this is the cwd: '${process.cwd()}'`),
   ).toEqual(`this is the cwd:  [project_dir]\\???(${process.cwd().length})`);
   expect(
-    telemetry.sanitizeMessage(`uppercase: '${process.cwd().toUpperCase()}'`),
+    sanitizeMessage(`uppercase: '${process.cwd().toUpperCase()}'`),
   ).toEqual(`uppercase:  [project_dir]\\???(${process.cwd().length})`);
   expect(
-    telemetry.sanitizeMessage(`lowercase: '${process.cwd().toLowerCase()}'`),
+    sanitizeMessage(`lowercase: '${process.cwd().toLowerCase()}'`),
   ).toEqual(`lowercase:  [project_dir]\\???(${process.cwd().length})`);
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: '${process.cwd()}' and something else`,
     ),
   ).toEqual(
@@ -37,7 +38,7 @@ test('Sanitize message, project_dir', () => {
     })  and something else`,
   );
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: ${process.cwd()} and something else`,
     ),
   ).toEqual(
@@ -49,7 +50,7 @@ test('Sanitize message, project_dir', () => {
 
 test('Sanitize message, node_modules', () => {
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: '${process.cwd()}\\node_modules'`,
     ),
   ).toEqual(
@@ -58,27 +59,27 @@ test('Sanitize message, node_modules', () => {
     })`,
   );
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: '${process.cwd()}\\node_modules\\foo'`,
     ),
   ).toEqual('this is the cwd:  node_modules\\foo');
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `uppercase: '${process.cwd().toUpperCase()}\\NODE_MODULES\\foo'`,
     ),
   ).toEqual('uppercase:  node_modules\\foo');
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `lowercase: '${process.cwd().toLowerCase()}\\NODE_MODULES\\'`,
     ),
   ).toEqual('lowercase:  node_modules\\');
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `trailing: '${process.cwd()}\\node_modules\\' and something else`,
     ),
   ).toEqual('trailing:  node_modules\\  and something else');
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: ${process.cwd()}\\node_modules and something else that could be part of the path`,
     ),
   ).toEqual(
@@ -90,7 +91,7 @@ test('Sanitize message, node_modules', () => {
     })`,
   );
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `this is the cwd: ${process.cwd()}\\node_modules\\ a file under nm`,
     ),
   ).toEqual(`this is the cwd: node_modules\\ a file under nm`);
@@ -98,15 +99,15 @@ test('Sanitize message, node_modules', () => {
 
 test('Sanitize message, other path', () => {
   expect(
-    telemetry.sanitizeMessage(`this is another path: 'A:\\foo\\bar\\baz'`),
+    sanitizeMessage(`this is another path: 'A:\\foo\\bar\\baz'`),
   ).toEqual(`this is another path:  [path]`);
 
   expect(
-    telemetry.sanitizeMessage(`this is another path: A:\\foo\\bar\\baz`),
+    sanitizeMessage(`this is another path: A:\\foo\\bar\\baz`),
   ).toEqual(`this is another path: [path]`);
 
   expect(
-    telemetry.sanitizeMessage(
+    sanitizeMessage(
       `Cannot find module 'react-native/package.json'
       Require stack:
       - ${process.env.APPDATA}\\npm-cache\\_npx\\1384\\node_modules\\react-native-windows-init\\lib-commonjs\\Cli.js
@@ -136,7 +137,7 @@ test('Sanitize stack frame', () => {
     assembly: 'asdf',
     line: 0,
   };
-  telemetry.sanitizeFrame(emptyFrame);
+  sanitizeFrame(emptyFrame);
   expect(emptyFrame).toEqual({
     level: 0,
     assembly: '',
@@ -152,7 +153,7 @@ test('Sanitize stack frame', () => {
     level: 0,
     line: 0,
   };
-  telemetry.sanitizeFrame(frame1);
+  sanitizeFrame(frame1);
   expect(frame1).toEqual({
     assembly: '',
     fileName: 'telemetry\\foo.js',
@@ -168,7 +169,7 @@ test('Sanitize stack frame', () => {
     level: 1,
     line: 42,
   };
-  telemetry.sanitizeFrame(frame2);
+  sanitizeFrame(frame2);
   expect(frame2).toEqual({
     assembly: '',
     fileName: 'telemetry\\foo.js',
@@ -179,13 +180,13 @@ test('Sanitize stack frame', () => {
 });
 
 test('basic setup', () => {
-  expect(telemetry.telClient.commonProperties.sessionId).toBeDefined();
+  expect(Telemetry.client!.commonProperties.sessionId).toBeDefined();
 
   expect(
-    telemetry.telClient.commonProperties.sessionId.length,
+    Telemetry.client!.commonProperties.sessionId.length,
   ).toBeGreaterThanOrEqual(32);
 
-  expect(telemetry.telClient.commonProperties.isTest).toEqual('true');
+  expect(Telemetry.client!.commonProperties.isTest).toEqual('true');
 });
 
 function b(s: string) {
@@ -197,7 +198,7 @@ function a(s: string) {
 
 test('thrown exception a->b, hello world', done => {
   let pass = false;
-  telemetry.telClient.addTelemetryProcessor((envelope, _) => {
+  Telemetry.client!.addTelemetryProcessor((envelope, _) => {
     if (envelope.data.baseType === 'ExceptionData') {
       const data = (envelope.data as any).baseData;
       expect(data.exceptions).toBeDefined();
@@ -222,20 +223,20 @@ test('thrown exception a->b, hello world', done => {
   try {
     a('world');
   } catch (e) {
-    telemetry.telClient.trackException({exception: e});
+    Telemetry.client!.trackException({exception: e});
   }
-  telemetry.telClient.flush();
+  Telemetry.client!.flush();
 
   expect(pass).toBeTruthy();
-  telemetry.telClient.clearTelemetryProcessors();
-  telemetry.telClient.addTelemetryProcessor(telemetry.sanitizeEnvelope);
+  Telemetry.client!.clearTelemetryProcessors();
+  Telemetry.client!.addTelemetryProcessor(sanitizeEnvelope);
   done();
 });
 
 test('thrown exception a->b, hello path', done => {
   let pass = false;
 
-  telemetry.telClient.addTelemetryProcessor((envelope, _) => {
+  Telemetry.client!.addTelemetryProcessor((envelope, _) => {
     if (envelope.data.baseType === 'ExceptionData') {
       const data = (envelope.data as any).baseData;
       expect(data.exceptions).toBeDefined();
@@ -262,27 +263,27 @@ test('thrown exception a->b, hello path', done => {
   try {
     a(process.cwd());
   } catch (e) {
-    telemetry.telClient.trackException({exception: e});
+    Telemetry.client!.trackException({exception: e});
   }
-  telemetry.telClient.flush();
+  Telemetry.client!.flush();
 
   expect(pass).toBeTruthy();
   expect(pass).toBeTruthy();
-  telemetry.telClient.clearTelemetryProcessors();
-  telemetry.telClient.addTelemetryProcessor(telemetry.sanitizeEnvelope);
+  Telemetry.client!.clearTelemetryProcessors();
+  Telemetry.client!.addTelemetryProcessor(sanitizeEnvelope);
   done();
 });
 
 test('trackEvent should not identify roleInstance', () => {
-  telemetry.telClient.addTelemetryProcessor((envelope, _) => {
+  Telemetry.client!.addTelemetryProcessor((envelope, _) => {
     expect(envelope.tags['ai.cloud.roleInstance']).toBeUndefined();
     return true;
   });
-  telemetry.telClient.trackEvent({
+  Telemetry.client!.trackEvent({
     name: 'test',
     properties: {},
   });
-  telemetry.telClient.flush();
-  telemetry.telClient.clearTelemetryProcessors();
-  telemetry.telClient.addTelemetryProcessor(telemetry.sanitizeEnvelope);
+  Telemetry.client!.flush();
+  Telemetry.client!.clearTelemetryProcessors();
+  Telemetry.client!.addTelemetryProcessor(sanitizeEnvelope);
 });

--- a/packages/@react-native-windows/telemetry/src/test/sanitize.test.ts
+++ b/packages/@react-native-windows/telemetry/src/test/sanitize.test.ts
@@ -5,7 +5,12 @@
  * @format
  */
 
-import {Telemetry, sanitizeMessage, sanitizeEnvelope, sanitizeFrame} from '../telemetry';
+import {
+  Telemetry,
+  sanitizeMessage,
+  sanitizeEnvelope,
+  sanitizeFrame,
+} from '../telemetry';
 import * as appInsights from 'applicationinsights';
 import {basename} from 'path';
 
@@ -19,9 +24,9 @@ test('Sanitize message, no-op', () => {
 });
 
 test('Sanitize message, project_dir', () => {
-  expect(
-    sanitizeMessage(`this is the cwd: '${process.cwd()}'`),
-  ).toEqual(`this is the cwd:  [project_dir]\\???(${process.cwd().length})`);
+  expect(sanitizeMessage(`this is the cwd: '${process.cwd()}'`)).toEqual(
+    `this is the cwd:  [project_dir]\\???(${process.cwd().length})`,
+  );
   expect(
     sanitizeMessage(`uppercase: '${process.cwd().toUpperCase()}'`),
   ).toEqual(`uppercase:  [project_dir]\\???(${process.cwd().length})`);
@@ -29,18 +34,14 @@ test('Sanitize message, project_dir', () => {
     sanitizeMessage(`lowercase: '${process.cwd().toLowerCase()}'`),
   ).toEqual(`lowercase:  [project_dir]\\???(${process.cwd().length})`);
   expect(
-    sanitizeMessage(
-      `this is the cwd: '${process.cwd()}' and something else`,
-    ),
+    sanitizeMessage(`this is the cwd: '${process.cwd()}' and something else`),
   ).toEqual(
     `this is the cwd:  [project_dir]\\???(${
       process.cwd().length
     })  and something else`,
   );
   expect(
-    sanitizeMessage(
-      `this is the cwd: ${process.cwd()} and something else`,
-    ),
+    sanitizeMessage(`this is the cwd: ${process.cwd()} and something else`),
   ).toEqual(
     `this is the cwd: [project_dir]\\???(${
       (process.cwd() + ' and something else').length
@@ -50,18 +51,14 @@ test('Sanitize message, project_dir', () => {
 
 test('Sanitize message, node_modules', () => {
   expect(
-    sanitizeMessage(
-      `this is the cwd: '${process.cwd()}\\node_modules'`,
-    ),
+    sanitizeMessage(`this is the cwd: '${process.cwd()}\\node_modules'`),
   ).toEqual(
     `this is the cwd:  [project_dir]\\???(${
       (process.cwd() + '\\node_modules').length
     })`,
   );
   expect(
-    sanitizeMessage(
-      `this is the cwd: '${process.cwd()}\\node_modules\\foo'`,
-    ),
+    sanitizeMessage(`this is the cwd: '${process.cwd()}\\node_modules\\foo'`),
   ).toEqual('this is the cwd:  node_modules\\foo');
   expect(
     sanitizeMessage(
@@ -98,13 +95,13 @@ test('Sanitize message, node_modules', () => {
 });
 
 test('Sanitize message, other path', () => {
-  expect(
-    sanitizeMessage(`this is another path: 'A:\\foo\\bar\\baz'`),
-  ).toEqual(`this is another path:  [path]`);
+  expect(sanitizeMessage(`this is another path: 'A:\\foo\\bar\\baz'`)).toEqual(
+    `this is another path:  [path]`,
+  );
 
-  expect(
-    sanitizeMessage(`this is another path: A:\\foo\\bar\\baz`),
-  ).toEqual(`this is another path: [path]`);
+  expect(sanitizeMessage(`this is another path: A:\\foo\\bar\\baz`)).toEqual(
+    `this is another path: [path]`,
+  );
 
   expect(
     sanitizeMessage(

--- a/packages/react-native-windows-init/src/Cli.ts
+++ b/packages/react-native-windows-init/src/Cli.ts
@@ -16,7 +16,7 @@ import * as chalk from 'chalk';
 // @ts-ignore
 import * as Registry from 'npm-registry';
 
-import {telemetryClient, isMSFTInternal} from '@react-native-windows/telemetry';
+import {Telemetry, isMSFTInternal} from '@react-native-windows/telemetry';
 
 /**
  * Important:
@@ -152,7 +152,9 @@ if (!argv.telemetry || process.env.AGENT_NAME) {
   if (argv.verbose) {
     console.log('Disabling telemetry');
   }
-  telemetryClient.config.disableAppInsights = true;
+  Telemetry.disable();
+} else {
+  Telemetry.setup();
 }
 
 function getReactNativeProjectName(): string {
@@ -415,7 +417,7 @@ function getRNWInitVersion(): string {
 
 function setExit(exitCode: ExitCode, error?: String): void {
   if (!process.exitCode || process.exitCode === ExitCode.SUCCESS) {
-    telemetryClient.trackEvent({
+    Telemetry.client?.trackEvent({
       name: 'init-exit',
       properties: {
         durationInSecs: process.uptime(),
@@ -594,6 +596,6 @@ function isProjectUsingYarn(cwd: string): boolean {
     }
     setExit(exitCode, error.message);
   } finally {
-    telemetryClient.flush();
+    Telemetry.client?.flush();
   }
 })();

